lib/semantics/resolve-names.cc

```diff
@@ -3672,8 +3676,62 @@ bool DeclarationVisitor::HandleUnrestrictedSpecificIntrinsicFunction(
   }
 }
 
+bool DeclarationVisitor::PassesLocalityChecks(
+    const parser::Name &name, Symbol &symbol) {
+  if (!IsVariableName(symbol)) {
+    SayLocalMustBeVariable(name, symbol);  // C1124
+    return false;
+  }
+  if (IsAllocatable(symbol)) {  // C1128
+    SayWithDecl(name, symbol,
+        "ALLOCATABLE variable '%s' not allowed in a locality-spec"_err_en_US);
+    return false;
+  }
+  if (IsOptional(symbol)) {  // C1128
+    SayWithDecl(name, symbol,
+        "OPTIONAL argument '%s' not allowed in a locality-spec"_err_en_US);
+    return false;
+  }
+  if (IsIntentIn(symbol)) {  // C1128
+    SayWithDecl(name, symbol,
+        "INTENT IN argument '%s' not allowed in a locality-spec"_err_en_US);
+    return false;
+  }
+  if (IsFinalizable(symbol)) {  // C1128
+    SayWithDecl(name, symbol,
+        "Finalizable variable '%s' not allowed in a locality-spec"_err_en_US);
+    return false;
+  }
+  if (IsCoarray(symbol)) {  // C1128
+    SayWithDecl(
+        name, symbol, "Coarray '%s' not allowed in a locality-spec"_err_en_US);
+    return false;
+  }
+  const DeclTypeSpec *type{symbol.GetType()};
+  if (type) {
```

Comment:
Maybe simply `if (const DeclTypeSpec *type{symbol.GetType()}) {` ?

Reply:
Thanks.  I'll make that change.